src/cursor/cur_version.c

```diff
@@ -0,0 +1,453 @@
+                 * particular version of the update.
+                 */
+                __wt_cursor_set_key(cursor, upd->txnid, upd->start_ts, upd->durable_ts,
+                  version_cursor->upd_stop_txnid, version_cursor->upd_stop_ts,
+                  version_cursor->upd_durable_stop_ts, upd->type, upd->prepare_state, upd->flags,
+                  WT_VERSION_UPDATE_CHAIN);
+
+                /*
+                 * Copy the update value into the version cursor as we don't know the value format.
+                 * If the update is a modify, reconstruct the value.
+                 */
+                if (upd->type != WT_UPDATE_MODIFY)
+                    __wt_upd_value_assign(cbt->upd_value, upd);
+                else
+                    WT_ERR(
+                      __wt_modify_reconstruct_from_upd_list(session, cbt, upd, cbt->upd_value));
+
+                cursor->value.data = cbt->upd_value->buf.data;
+                cursor->value.size = cbt->upd_value->buf.size;
+                F_SET(cursor, WT_CURSTD_VALUE_INT);
+
+                upd_found = true;
+                version_cursor->next_upd = upd->next;
+            }
+        }
+    }
+
+    if (!upd_found && !F_ISSET(version_cursor, WT_VERSION_CUR_ON_DISK_EXHAUSTED)) {
+        /* If the key is on an insert list only, there is no ondisk value nor history store value.
+         */
+        if (cbt->ins || WT_COL_APPEND(page) != NULL) {
+            F_SET(version_cursor, WT_VERSION_CUR_ON_DISK_EXHAUSTED);
+            F_SET(version_cursor, WT_VERSION_CUR_HS_EXAUSTED);
+        } else {
+            page = cbt->ref->page;
+
+            /* Retrieve the value for each type of underlying table structure. */
+            switch (page->type) {
+            case WT_PAGE_ROW_LEAF:
+                rip = &page->pg_row[cbt->slot];
+                __wt_row_leaf_value_cell(session, cbt->ref->page, rip, vpack);
+                break;
+            case WT_PAGE_COL_FIX:
+                break;
+            case WT_PAGE_COL_VAR:
+                recno = cbt->ref->ref_recno;
+                cip = &page->pg_var[cbt->slot];
+                cell = WT_COL_PTR(page, cip);
+                __wt_cell_unpack_kv(session, page->dsk, cell, vpack);
+                break;
+            default:
+                WT_ERR(__wt_illegal_value(session, page->type));
+                break;
+            }
+
+            if (!WT_TIME_WINDOW_HAS_STOP(&vpack->tw)) {
+                durable_stop_ts = version_cursor->upd_durable_stop_ts;
+                stop_ts = version_cursor->upd_stop_ts;
+                stop_txn = version_cursor->upd_stop_txnid;
+            } else {
+                durable_stop_ts = vpack->tw.durable_stop_ts;
+                stop_ts = vpack->tw.stop_ts;
+                stop_txn = vpack->tw.stop_txn;
+            }
+
+            __wt_cursor_set_key(cursor, vpack->tw.start_txn, vpack->tw.start_ts,
+              vpack->tw.durable_start_ts, stop_txn, stop_ts, durable_stop_ts, 0, 0, 0,
```

Comment:
Should set the type as standard. I'm a bit concerned directly exposing the internal wiredtiger information such as type and prepare state. We probably should define them separately in the api and convert the internal value to the api value and then return to the user. If we directly return them, their value may change and it will break a lot of tests.

Reply:
We should return prepare state here as well because we may write prepared to disk.